web/blueprints/finance/__init__.py

```diff
@@ -608,6 +610,114 @@ def transactions_show_json(transaction_id):
+        'finance/transactions_unconfirmed.html',
+        page_title="Unbestätigte Transaktionen",
+        unconfirmed_transactions_table=UnconfirmedTransactionsTable(
+            data_url=url_for(".transactions_unconfirmed_json"))
+    )
+
+
+@bp.route('/transactions/unconfirmed/json')
+def transactions_unconfirmed_json():
+    transactions = Transaction.q.filter_by(confirmed=False).order_by(Transaction.posted_at).limit(100).all()
+
+    return jsonify(
+        items=[
+        {
+            'description': {
+                'href': url_for(".transactions_show", transaction_id=transaction.id),
+                'title': transaction.description,
+                'new_tab': True
+            },
+            'author': {
+                'href': url_for("user.user_show",
+                                user_id=transaction.author.id),
+                'title': transaction.author.name,
+                'new_tab': True
+            },
+            'date': date_format(transaction.posted_at),
+            'amount': money_filter(transaction.amount),
+            'details': {
+                'href': url_for(".transactions_show", transaction_id=transaction.id),
+                'title': 'Details',
+                'btn_class': 'btn-primary',
+                'new_tab': True
+            },
+            'actions': [{
+                    'href': url_for(".transaction_confirm",
+                                    transaction_id=transaction.id),
+                    'title': 'Bestätigen',
+                    'icon': 'glyphicon-ok'
+                },{
+                    'href': url_for(".transaction_delete",
+                                    transaction_id=transaction.id),
+                    'title': 'Löschen',
+                    'icon': 'glyphicon-trash'
+                }
+            ],
+        } for transaction in transactions])
+
+
+@bp.route('/transaction/<int:transaction_id>/confirm', methods=['GET', 'POST'])
+@access.require('finance_change')
+def transaction_confirm(transaction_id):
+    transaction = Transaction.q.get(transaction_id)
+
+    if transaction is None:
+        flash(u"Transaktion existiert nicht.", 'error')
+        abort(404)
+
+    if transaction.confirmed:
+        flash(u"Diese Transaktion wurde bereits bestätigt.", 'error')
+        abort(404)
+
+    lib.finance.transaction_confirm(transaction)
+
+    session.commit()
+
+    flash(u'Transaktion bestätigt.', 'success')
+    return redirect(url_for('.transactions_unconfirmed'))
+
+
+@bp.route('/transaction/<int:transaction_id>/delete', methods=['GET', 'POST'])
+@access.require('finance_change')
+def transaction_delete(transaction_id):
+    transaction = Transaction.q.get(transaction_id)
+
+    if transaction is None:
+        flash(u"Transaktion existiert nicht.", 'error')
+        abort(404)
+
+    if transaction.confirmed:
+        flash(u"Diese Transaktion wurde bereits bestätigt und kann daher nicht gelöscht werden.", 'error')
+        abort(404)
```

Comment:
Also applies to the other 404 added in this file. How does the user perceive the 404 error?
Is he redirected to the unconfirmed transactions overview by ` 'cancel_to': url_for('.transactions_unconfirmed')`?

Reply:
This and other 4xx errors are very rare and are mostly only triggered when the url is changed manually. So there dont't has to be a fancy redirection. It just displays an error like here:

https://pycroft.agdsn.de/user/1111111/

The user can then just use the back button of its browser.

However, 404 is not the correct status code for this.